addons/survey/controllers/main.py

```diff
@@ -365,16 +383,22 @@ def survey_display_page(self, survey_token, answer_token, **post):
         return request.render('survey.survey_page_fill',
             self._prepare_survey_data(access_data['survey_sudo'], answer_sudo, **post))
 
-    @http.route('/survey/get_background_image/<string:survey_token>/<string:answer_token>', type='http', auth="public", website=True, sitemap=False)
-    def survey_get_background(self, survey_token, answer_token):
-        access_data = self._get_access_data(survey_token, answer_token, ensure_token=True)
-        if access_data['validity_code'] is not True:
-            return werkzeug.exceptions.Forbidden()
+    @http.route([
+        '/survey/get_background_image/<string:survey_token>/<int:question_id>',
+        '/survey/get_background_image/<string:survey_token>',
```

Comment:
Small side note: would order them the other way around, we usually start with the route with the least amount of arguments and then pile on.

Reply:
allez, vendu